pkg/plugins/golang/v3/commons.go

```diff
@@ -84,7 +84,7 @@ manifests: controller-gen`
 		}
 
 		// latest version of controller-runtime where v1beta1 is supported
-		const controllerRuntimeVersionForVBeta1 = "v0.9.2"
+		const controllerRuntimeVersionForVBeta1 = "v0.13.0"
```

Comment:
@oscr this one we cannot update.
The kubebuilder CLI has an option to create the projects using crd/webhooks API v1beta1 which is no longer provided since k8s 1.22. This logic will update the go.mod file to use the deps that works with v1beta1 if/when someone calls the command to create api [options] crd-version=v1beta1

Therefore, could you please revert it and also add a comment on the top to explain it so that others in the future does not try to update this one?

Example (something like):

````
// DO NOT UPDATE THIS VERSION
// Note that this implementation will update the go.mod to downgrade the versions for those that are
// compatible v1beta1 CRD/Webhooks k8s core APIs if/when an user tries to create a API with
// create api [options] crd-version=v1beta1. The flag/feature is deprecated. however, to ensure that backwards 
// compatible we must introduce this logic. Also, note that when we bump the k8s dependencies we need to
// ensure that the following replacements will be done accordingly to downgrade the versions. 
// The next version of the Golang base plugin (go/v4-alpha) no longer provide this feature.

Reply:
/hold

Reply:
Btw, we have been removing the testdata and samples with this option because it is deprecated. 
So, that it did not fail in the tests but we need to ensure that we have a testdata that checks it.
The ci should fail with this change. I am adding an issue so we are able to ensure that.